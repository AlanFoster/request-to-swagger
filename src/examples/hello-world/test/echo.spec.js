import echoSchema from '../echo';
import searchedRepositoriesSchema from "../../github/search-repositories";

describe('echo', function () {
  it('generates a valid schema', function () {
    expect(echoSchema).toMatchSnapshot();
    expect(searchedRepositoriesSchema).toBeValidSwagger();
  });
});
