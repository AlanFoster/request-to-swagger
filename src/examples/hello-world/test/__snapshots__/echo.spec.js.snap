// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`echo generates a valid schema 1`] = `
Object {
  "info": Object {
    "title": "Generated Schema",
    "version": "1.0",
  },
  "paths": Object {
    "/echo/hello": Object {
      "get": Object {
        "consumes": Array [],
        "parameters": Array [],
        "produces": Array [
          "application/json",
        ],
        "responses": Object {
          "200": Object {
            "description": "OK",
            "schema": Object {
              "properties": Object {
                "response": Object {
                  "type": "string",
                },
              },
              "required": Array [
                "response",
              ],
              "type": "object",
            },
          },
        },
      },
    },
  },
  "swagger": "2.0",
}
`;
